staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -652,6 +657,33 @@ func (f *DeltaFIFO) Replace(list []interface{}, _ string) error {
 		}
 	}
 
+	// detect deletions for items in the queue that are not yet present in knownObjects
+	for k, oldItem := range f.items {
```

Comment:
Figured out we should probably do that at the same time as this to make the behavior consistent and to avoid more ambiguities. Pushed a new set of commits to do it.